### PR TITLE
refactor: rename recommended config to dom

### LINKED ISSUE #182: Rename `recommended` config
Motivated by [this comment](https://github.com/testing-library/eslint-plugin-testing-library/pull/177#discussion_r442950929) I want to propose renaming `recommended` config to `dom`.

I picked that name because is the usual config included in an ESLint plugin, but it doesn't look like proper name and seems misleading. I have seen some people too configuring `recommended` + `react` or another framework, so the name can be definitely improved.

My proposal then is to rename `recommended` to `dom` so 1) it's obvious it's not a recommended one in general, but specific one for DOM Testing Library and 2) it makes clearer the fact you don't need to enable that one together with another one for a Testing Library framework.

I can address this one myself tomorrow (including doc updates) and add it to v4, but I'd like to know your opinion first.

### PR DESCRIPTION
**Pull request by Belco90**
BREAKING CHANGE: recommended config preset has been renamed to dom, so make sure you update it in your ESLint config file.

Closes #182

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -68,37 +68,25 @@ Then configure the rules you want to use under the rules section.
 
 ## Shareable configurations
 
-### Recommended
+This plugin exports several recommended configurations that enforces good practices for specific Testing Library packages.
```
Comment (nickmccurdy) [inline README.md:71]:
```suggestion
This plugin exports several recommended configurations that enforce good practices for specific Testing Library packages.
```

Comment (Belco90) [reply README.md]: Thanks.

**Comment by Belco90** [comment]
> LGTM, one tiny nit.
> 
> I think we need to do something about the `dom` badge 😅
> image

Oops I didn't realize about it haha. Let me see what I can come up with to improve this.

**Comment by Belco90** [comment]
What do you think now? 😅 

image

**Comment by timdeschryver** [comment]
🤩 Much better! 👍👍

**Pull request merged**